src/components/PetEditPage.jsx

```diff
@@ -0,0 +1,167 @@
+import { useState, useEffect } from 'react'
+import axios from 'axios'
+import '../styles/style.css'
+import { useNavigate } from 'react-router-dom'
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import usePets from '../hooks/usePets'
+import { useAuth } from '../hooks/store'
+
+const apiUrl = import.meta.env.VITE_API_BACKEND_URL + "/api/v1/user/pet/edit";
+
+//TODO make "select pet" unsubmitable
+
+const PetEditPage = () => {
+    const { pets } = usePets(); // Load pets from your custom hook
+    const [selectedPetIndex, setSelectedPetIndex] = useState(null); // Index of the selected pet
+    const [formData, setFormData] = useState({
+        petName: '',
+        petBreed: '',
+        petSex: '',
+        petBirthDate: '',
+        lastCheckUp: '',
+        nextCheckUp: '',
+        specialCondition: ''
+    })
+
+    const navigate = useNavigate()
+    const { token } = useAuth()
+
+    // Handle when a user selects a pet from the dropdown
+    const handlePetSelect = (e) => {
+        const index = e.target.value;
+        setSelectedPetIndex(index);
+
+        if (index !== '') {
+            const pet = pets[index];
+            // Set form data to the selected pet's data
+            setFormData({
+                petName: pet.petName || '',
+                petBreed: pet.petBreed || '',
+                petSex: pet.petSex || '',
+                petBirthDate: pet.petBirthDate || '',
+                lastCheckUp: pet.lastCheckUp || '',
+                nextCheckUp: pet.nextCheckUp || '',
+                specialCondition: pet.specialCondition || ''
+            });
+        } else {
+            // Reset form data if no pet is selected
+            setFormData({
+                petName: '',
+                petBreed: '',
+                petSex: '',
+                petBirthDate: '',
+                lastCheckUp: '',
+                nextCheckUp: '',
+                specialCondition: ''
+            });
+        }
+    }
+
+    const handleChange = (e) => {
+        const { name, value } = e.target;
+        setFormData({
+            ...formData,
+            [name]: value
+        });
+    }
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+
+        // Ensure a pet is selected
+        if (selectedPetIndex === null) {
+            toast.error('Please select a pet to edit');
+            return;
+        }
+
+        const selectedPetId = pets[selectedPetIndex].id; // Assuming each pet has a unique `id`
+
+        axios.patch(`${apiUrl}/${selectedPetId}`, formData, // Send the selected pet ID in the API call
+            {
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': `Bearer ${token}`
+                }
+            })
+            .then(() => {
+                toast.success('Pet Changed successfully');
+                setTimeout(() => navigate(`/user/home`), 1500);
```

Comment:
With the latest changes there are no need to set timeout `toast` can be used globally.